### PR TITLE
Add Aplus App

### PR DESCRIPTION
**Pull request by natanfelles**
Adding a new framework: [Aplus App](https://github.com/aplus-framework/app).

Thank you!

### CODE REVIEW

File: php/aplus-app/config.yaml [hunk 1]
```diff
@@ -0,0 +1,8 @@
+framework:
+  github: aplus-framework/app
+  version: 23.3
+  website: aplus-framework.com/packages/app
+
+bootstrap:
```
Comment (waghanza) [inline php/aplus-app/config.yaml:6]:
```suggestion
   bootstrap:
```

**Review by waghanza — CHANGES_REQUESTED** [review]
I think you need some extensions, `php_ext`

**Comment by natanfelles** [comment]
Hi @waghanza !

Please could you tell me which file should I put the PHP extensions in and an example of that?

Thank you very much!

**Comment by waghanza** [comment]
In config.yaml @natanfelles, see other config.yaml files, or list me PHP extension required, I'll suggest something

File: php/aplus-app/config.yaml [hunk 2]
```diff
@@ -0,0 +1,8 @@
+framework:
+  github: aplus-framework/app
+  version: 23.3
+  website: aplus-framework.com/packages/app
+
+  bootstrap:
+    - mkdir -p storage/logs
+    - chown -R www-data:www-data storage/logs
```
Comment (waghanza) [inline php/aplus-app/config.yaml:8]:
```suggestion
    - chown -R www-data:www-data storage/logs
    
    deps:
    - libsodium
    
    modules:
    - sodium
    
    
```

**Review by waghanza — CHANGES_REQUESTED** [review]

File: php/aplus-app/config.yaml (hunk 2 above)
Comment (waghanza) [inline php/aplus-app/config.yaml:8]:
```suggestion
framework:
  github: aplus-framework/app
  version: 23.3
  website: aplus-framework.com/packages/app

  bootstrap:
    - mkdir -p storage/logs
    - chown -R www-data:www-data storage/logs
  deps:
    - libsodium
    
  modules:
    - sodium
```

**Review by waghanza — CHANGES_REQUESTED** [review]

File: php/aplus-app/config.yaml (hunk 2 above)
Comment (natanfelles) [reply php/aplus-app/config.yaml:8]: Here are the extensions we use: https://gitlab.com/aplus-framework/images/base/-/blob/3/Dockerfile#L15-33

File: php/aplus-app/config.yaml [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  deps:
+    - libsodium
```
Comment (waghanza) [inline php/aplus-app/config.yaml:10]:
```suggestion
    - libsodium-dev
```

**Review by waghanza — CHANGES_REQUESTED** [review]

File: php/aplus-app/config.yaml [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  website: aplus-framework.com/packages/app
+
+  bootstrap:
+    - mkdir -p storage/logs
+    - chown -R www-data:www-data storage/logs
+  deps:
+    - libsodium-dev
+    
+  modules:
+    - sodium
```
Comment (waghanza) [inline php/aplus-app/config.yaml:13]:
```suggestion
framework:
  github: aplus-framework/app
  version: 23.3
  website: aplus-framework.com/packages/app

  bootstrap:
    - mkdir -p storage/logs
    - chown -R www-data:www-data storage/logs
  deps:
    - libsodium-dev
    - libmemcached-dev
    - yaml-dev

  modules:
    - sodium
    - mysqli

  extensions:
  - igbinary
  - memcached
  - msgpack
  - redis
  - yaml
  - intl
```

**Comment by waghanza** [comment]
Not sure I understand this compilation error
https://github.com/the-benchmarker/web-frameworks/actions/runs/6025315294/job/16345785190?pr=6653#step:7:4154

@natanfelles @sy-records

File: php/aplus-app/config.yaml [hunk 5]
```diff
@@ -0,0 +1,24 @@
...
+  version: 23.3
+  website: aplus-framework.com/packages/app
+
+  bootstrap:
+    - mkdir -p storage/logs
+    - chown -R www-data:www-data storage/logs
+  deps:
+    - libsodium-dev
+    - libmemcached-dev
+    - yaml-dev
```
Comment (sy-records) [inline php/aplus-app/config.yaml:12]:
```suggestion
    - yaml-dev
    - libicu-dev
```

**Review by sy-records — CHANGES_REQUESTED** [review]

File: php/aplus-app/config.yaml (hunk 5 above)
Comment (waghanza) [reply php/aplus-app/config.yaml:12]:
```suggestion
    - yaml-dev
    - icu-dev
```

**Comment by waghanza** [comment]
ℹ️ libicu-dev is for debian, we are on alpine

**Comment by sy-records** [comment]
intl Doesn't seem to be able to install using pecl anymore. https://github.com/the-benchmarker/web-frameworks/actions/runs/6033838164/job/16371162804?pr=6653#step:7:3348

> This package is not maintained anymore and has been superseded. Package has moved to channel http://php.net/manual/en/book.intl.php

**Comment by waghanza** [comment]
Same @natanfelles https://github.com/the-benchmarker/web-frameworks/actions/runs/6025315294/job/16345783452#step:7:4176

File: php/aplus-app/config.yaml [hunk 6]
```diff
@@ -0,0 +1,25 @@
...
+    - sodium
+    - mysqli
+
+  extensions:
+  - igbinary
+  - memcached
+  - msgpack
+  - redis
+  - yaml
+  - intl
```
Comment (sy-records) [inline php/aplus-app/config.yaml:25]:
```suggestion
  modules:
    - sodium
    - mysqli
    - intl

  extensions:
  - igbinary
  - memcached
  - msgpack
  - redis
  - yaml
```

**Review by sy-records — CHANGES_REQUESTED** [review]

File: php/aplus-app/config.yaml [hunk 7]
```diff
@@ -0,0 +1,25 @@
...
+  deps:
+    - libsodium-dev
+    - libmemcached-dev
+    - yaml-dev
+    - icu-dev
+
+  modules:
+    - sodium
+    - mysqli
+    - intl
```
Comment (sy-records) [inline php/aplus-app/config.yaml:18]:
```suggestion
    - intl
    - gd
```
Need require ext-gd

**Review by waghanza — APPROVED** [review]

**Review by sy-records — APPROVED** [review]

File: php/aplus-app/composer.json [hunk 8]
```diff
@@ -0,0 +1,18 @@
+{
+    "require": {
+        "aplus/framework": "^23"
```
Comment (waghanza) [inline php/aplus-app/composer.json:3]:
```suggestion
        "aplus/framework": "~23.3.0"
```

**Review by waghanza — APPROVED** [review]

**Pull request merged**